discovery_engine/lib/src/domain/event_handler.dart

```diff
@@ -167,8 +166,6 @@ class EventHandler {
         response = await _feedManager.handleFeedClientEvent(clientEvent);
       } else if (clientEvent is DocumentClientEvent) {
         await _documentManager.handleDocumentClientEvent(clientEvent);
-      } else if (clientEvent is SystemClientEvent) {
-        // TODO: we need to handle other system events (config changed)
```

Comment:
We should handle the event here. We can keep an `Engine?` in the fields instead of `Future<Engine>?`.
For the `maxDoc` that `FeedManager` needs we can have an internal configuration object that we update here and that the `FeedManager` can use to read. This will be also used by the `ActiveSearchManager`.

Reply:
> await engine.requestFeed();

Or assign the engine returned from awaiting the future to a variable on line 151/152.

Reply:
but we have a variable around that we don't need.

Reply:
:thinking: Are we speaking about the same thing?

Like what fields do you mean?

Reply:
`_engineFuture`

Reply:
But we want to make `_engineFuture` a future to avoid any potential race conditions between init started and making a request before initialization completes.

Reply:
And that future will return the `Engine` instance, so if we need to use that anywhere we can just store it at l.151/152 in a variable.

Reply:
As a related but slightly off-topic side question: How are we meant to handle engine failure (after returning an error). Will we retry the initialization?

Reply:
> But we want to make `_engineFuture` a future to avoid any potential race conditions between init started and making a request before initialization completes.

We can do the same.
```dart
Engine? _engine;

... handleEvent(...) {
  if(event is Init) {
     _engine = await _initEngine(...);
  }
  
  if(_engine == null) return Error(EngineNotReady);
}
```
We should also throw an error if the engine is already initialized and the user try to init it again.

Reply:
The only difference that I see now is that now a message sent after `Init` will wait, while with this it will fail.
I don't see a reason to wait. Is up to the user to send us messages after he knows that the engine has been initialized correctly.

Reply:
> As a related, but slightly off-topic side question: How are we meant to handle engine failure (after returning an error). Will we retry the initialization?

Why we should reinit after an error? We don't really have anything that can be fixed by a reinit.

Reply:
We have nothing which automatically can be fixed, but I could imagine Team Blue having some way to change the config in dev builds, potentially in ways which could make it fail. 

Anyway they can then just send `Init` again, I guess.